bundle/edu.gemini.wdba.xmlrpc.server/src/main/java/edu/gemini/wdba/session/SessionXmlRpcHandler.java

```diff
@@ -72,8 +74,11 @@ private static void log(String method, String... args) {
     // this has to be set from the activator and then later used by the
     // SessionXmlRpcHandler that is eventually created.
     private static SessionManagement sm = null;
-    public static synchronized void setContext(WdbaContext context) {
-        sm = (context == null) ? null : new SessionManagement(context, new ProductionSessionConfiguration(context));
+    public static synchronized void setContext(
+        WdbaContext context,
+        Consumer<ExecEvent> eventConsumer
```

Comment:
The `eventConsumer` is configured depending upon whether we're in production or testing.  It receives an event and does something appropriate with it.